### DISCUSSION TITLE
Map time-slider

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#1650] [Ideas]
In my migration report I have a lot of long time series. Often starting in 2005. If I click on the map, data of the first year is shown. If you don't look at the time slider you may think the shown values are from the recent year. Especially because my tables have the recent year in the first row...

It would be great to preset the recent year... What do you think?
@MoBosse @LucyGwilliamAdmin @brockfanning 

Example indicator: https://christoph-lsn.github.io/MT_Site/1-2-3/

site repo; https://github.com/christoph-LSN/MT_Site
data repo: https://github.com/christoph-LSN/IM_Data

![image](https://user-images.githubusercontent.com/94443931/171658966-57ab0854-31f3-47d4-bbc8-5149ffe7e623.png)

**Comment by brockfanning** [comment]
This seems reasonable to me: starting with the last year instead of the first year - so that the slider's "handle" would start out all the way to the right.

**Comment by MoBosse** [comment]
Hi @christoph-LSN @brockfanning ,
I think this is exactly what I did on our national platform ([example here](https://sustainabledevelopment-deutschland.github.io/4-1-a/)) right?
Unfortunately this is a very old open-sdg version so not good for a pr.

**Reply by brockfanning** [reply]
@MoBosse Do you happen to remember the bit of code that got the slider to start at the end? I'm having trouble figuring that out.

**Reply by MoBosse** [reply]
@brockfanning I dont't know if it will work in 2.0.0 but for 1.8.0 changing [this line](https://github.com/open-sdg/open-sdg/blob/1.8.0-dev/_includes/assets/js/plugins/leaflet.yearSlider.js#L125) to `currentTime: new Date(years.slice(-1)[0].time).getTime(),` and [this line](https://github.com/open-sdg/open-sdg/blob/1.8.0-dev/_includes/assets/js/plugins/jquery.sdgMap.js#L414)
to `plugin.currentYear = plugin.years.slice(-1)[0];` makes the map starting with the latest year available.
Only the slider itself still starts on the left side:
![grafik](https://user-images.githubusercontent.com/50657629/172781120-3e1b0d37-1942-4afb-819f-ff27b767ed8a.png)

**Reply by brockfanning** [reply]
@MoBosse Thanks - I'll check on that. I'm also having a hard time getting the slider to appear on the right, even though the year and data were correctly reflecting the most recent year.

**Reply by christoph-LSN** [reply]
@MoBosse @brockfanning I have tried it like MoBosse described - but it does not work. The slider shows the recent year but moves anymore. The legend and the change breakdown button have not been shown anymore...

I created a PR https://github.com/christoph-LSN/MT_Site/pull/3

Example Indicator: https://christoph-lsn.github.io/MT_Site/1-2-3/

![image](https://user-images.githubusercontent.com/94443931/172905290-4502f40a-134e-4359-ac76-fea740e6bf13.png)

**Reply by christoph-LSN** [reply]
@brockfanning @MoBosse 
I tried it again and changed 

[jquery.sdgMaps.js](https://github.com/christoph-LSN/MT_Site/blob/c39b8334fab01f05c880411bc806f60a3fba3865/_includes/assets/js/plugins/jquery.sdgMap.js#L473)

and 

[leaflet.yearSlider.js](https://github.com/christoph-LSN/MT_Site/blob/c39b8334fab01f05c880411bc806f60a3fba3865/_includes/assets/js/plugins/leaflet.yearSlider.js#L125)

like @MoBosse descibed and it seems to work well:

Example Indicator: [2-1-1](https://christoph-lsn.github.io/MT_Site/2-1-1/)

May be the errors last time came from the work in progress with the leaflet map disaggregation things....

**Comment by christoph-LSN** [comment]
@MoBosse good job! especially the time-slider... Showing the recent year and start sliding from the first year.... But perhaps it would be better to stop sliding when reaching the recent year.

**Comment by brockfanning** [comment]
Related to this - we were discussing internally and @phillipgardiner brought up the point that this may also make sense for the table, which could be sorted by year _descending_ instead of ascending.

**Reply by christoph-LSN** [reply]
@brockfanning

we discussed this last year: https://github.com/open-sdg/open-sdg/discussions/1399

My tables are sorted descending: https://christoph-lsn.github.io/MT_Site/1-1-1/

**Comment by brockfanning** [comment]
As of 2.1.0 the maps should be in descending order, so I'll mark this answered. EDIT: I guess I can't mark it answered - but this should be resolved with 2.1.0.
